jupyterhub/singleuser/mixins.py

```diff
@@ -366,6 +401,13 @@ def _notebook_dir_validate(self, proposal):
             raise TraitError("No such notebook dir: %r" % value)
         return value
 
+    @default('log_level')
+    def _log_level_defaul(self):
```

Comment:
```suggestion
    def _default_log_level(self):
```
Might as well match the naming convention unless there's areason to make it different?

Reply:
Yeah, fixed to match the convention. `_:trait_name:_default` is the convention from when traitlets only had magic method names, no `@default` decorators. The name no longer has any significance, but might as well use the established one. I think we could use `def _:(self)` for all of these if we wanted (not sure how it would like using the same name for all of them...)